ui/ui-components/contexts/api.tsx

```diff
@@ -0,0 +1,5 @@
+import { createContext, useContext } from "react";
+import type { Client } from "../client/client";
+
+export const ApiContext = createContext<{ client: Client }>(undefined);
+export const useApi = () => useContext(ApiContext);
```

Comment:
After looking at the code. I'm torn if this particular function should be in `/hooks/useApi` along the rest of the other context functions. The implementation of it as a context seems irrelevant when it's being used.

Reply:
This could potentially just go into `WebAppContext` but I'd have to find-replace both the useApi and the import and that would be a _lot_ of gruntwork for not much gain. This is a context, we should keep it here.